docs/reference/members.md

```diff
@@ -32,6 +32,26 @@ projects_and_groups:
       keep_bots: true
 ```
 
+### Custom Roles (GitLab Ultimate Only)
+Assigning of Custom Roles to Project `members` is supported within GitLabForm configuration.
+
+!!! warning
+
+    * user `access_level` MUST still be supplied and MUST match the base_access_level of the custom role
+
+```yaml
+projects_and_groups:
+  group_1/project_1:
+    members:
+      users:
+        my-user:
+          access_level: maintainer
+          member_role_id: 2
```

Comment:
The API expects an id but for usability purpose should we not allow the Name attribute of the role to be used and the id looked up from that?  Could be an issue with .com usage but for self-managed is probably preferable?

Reply:
I agree. It could be added in a separate PR if it'll make this PR too large.

Reply:
I'll look up how we do it with project names etc

Reply:
@adam-moss @amimas I've pushed a commit which should support names as well as Ids, the difficulty for resolving between the two is that GitLab SaaS will use a different API endpoint for custom roles to Self-Managed/Dedicated